File: Dev/C++/Mascaret/src/HAVE/VirtualHumanClass.cpp
#include "HAVE/VirtualHumanClass.h"
#include "HAVE/BodyClass.h"
namespace HAVE {

VirtualHumanClass::VirtualHumanClass(const string & name) : AgentClass(name)
{
	_bodyClass=shared_ptr<BodyClass>(new BodyClass(string("Body"+name),this));
}

VirtualHumanClass::~VirtualHumanClass()
{}

}

File: Dev/CS/Mascaret/Mascaret/HAVE/VirtualHumanClass.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class VirtualHumanClass : AgentClass
    {

        private BodyClass bodyClass;
        public BodyClass BodyClass
        {
            get { return bodyClass; }
            set { bodyClass = value; }
        }

        public VirtualHumanClass(string name)
            : base(name)
        {
            bodyClass = new BodyClass("Body_" + name, this);
        }

    }
}

